### PR TITLE
Add device id and some user properties

### PR DESCRIPTION
**Pull request by nathankim0**

### CODE REVIEW

File: Amplitude/Amplitude.cs [hunk 1]
```diff
@@ -1,9 +1,11 @@
 ﻿using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Net.Http;
 using System.Text;
 using System.Text.Json;
+using Xamarin.Essentials;
```
Comment (DenisNP) [inline Amplitude/Amplitude.cs:8]: This is universal library for any runtime. Please, do not use Xamarin or other platform-dependent namespaces.

File: Amplitude/Amplitude.cs [hunk 2]
```diff
@@ -69,17 +88,41 @@ public static void DisposeFor(string userId)
...
             {
                 throw new Exception("You should call Amplitude.Initialize(...) with your API_KEY first");
             }
 
             _userId = userId;
+            _deviceId = deviceId;
             _userProperties = userProperties;
         }
 
+        //private Amplitude(string deviceId, Dictionary<string, object> userProperties = null)
```
Comment (DenisNP) [inline Amplitude/Amplitude.cs:103]: Commented code can be removed, I suppose.

File: Amplitude/Amplitude.cs [hunk 3]
```diff
@@ -34,19 +53,19 @@ public static void Initialize(string apiKey)
         /// <param name="storeInstance">If enabled, InstanceFor will return same instance for same user</param>
         /// <param name="userProperties">Properties that should be added to user</param>
         /// <returns>Amplitude service instance</returns>
-        public static Amplitude InstanceFor(string userId, bool storeInstance = false, Dictionary<string, object> userProperties = null)
+        public static Amplitude InstanceFor(string userId, string deviceId, bool storeInstance = false, Dictionary<string, object> userProperties = null)
```
Comment (DenisNP) [inline Amplitude/Amplitude.cs:56]: I don't think that DeviceId is required property for all cases. E.g.: chat-bot, web application can not know about user device. So, at least this constructor should be in additional to the minimal one, without DeviceId.

**Review by DenisNP — CHANGES_REQUESTED** [review]
First of all, thank you for the PR!

I've added some comments, please, take a look

**Comment by nathankim0** [comment]
Thanks for the comment. As a junior developer, it has helped me a lot. I will close the pull request.

**Pull request closed without merging**